### ISSUE TITLE
Unable to retrieve info from stream mp3

### ISSUE THREAD

**Issue by jurgenhaas** [music-assistant/hass-music-assistant#707]
### What version of Music Assistant has the issue?

2022.7

### The problem

I have a list of streaming mp3 files that point to a local HDHomeRun and when I want to open that directory containing a number of such files, I receive this exception:

```
2022-07-23 11:00:31 ERROR (MainThread) [music_assistant.music.file] Error processing /media/music/hdhome/channel.mp3 - Unable to retrieve info for /media/music/hdhome/channel.mp3: Invalid argument
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/helpers/tags.py", line 208, in parse_tags
    raise InvalidDataError(error["string"])
music_assistant.models.errors.InvalidDataError: Invalid argument

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 234, in sync_library
    track = await self._parse_track(entry.path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 459, in _parse_track
    tags = await parse_tags(track_path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/helpers/tags.py", line 211, in parse_tags
    raise InvalidDataError(
music_assistant.models.errors.InvalidDataError: Unable to retrieve info for /media/music/hdhome/channel.mp3: Invalid argument
```

The mp3 files looks like this:

```
http://192.168.3.54:5004/auto/v11
```

This is how I play the radio stations on vlc and that also work with the media player in HA, but not with the music assistant. Is there a chance to either fix that or to integrate those channels into MA otherwise?

### How to reproduce

Create a channel.mp3 like the above and store it in a directory that's integrated into MA with the filesystem.

### What version of Home Assistant Core are your running

2022.7.6

### What type of installation are you running?

Home Assistant Container

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by marcelveldt** [comment]
How is this supposed to work ? An mp3 file used as playlist?!
Let's start by saying that it is even ridiculous that VLC plays that :-)

MA has a better solution for this.
Just create a playlist (file with m3u extension) and paste those lines in there.
That should work fine and if these are tracks you can add them to the library.

**Comment by jurgenhaas** [comment]
Thanks for your quick reply. So I renamed channel.mp3 into channel.m3u and I can now see the channel when going into that directory with MA. But when I select that channel, I receive another exception:

```
2022-07-23 12:17:28 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140159262078096] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 114, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 533, in websocket_playlist
    item = await mass.music.playlists.get(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/media_controller.py", line 162, in get
    details = await self.get_provider_item(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/media_controller.py", line 395, in get_provider_item
    item = await provider.get_item(self.media_type, item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 248, in get_item
    return await self.get_playlist(prov_item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 321, in get_playlist
    return await self._parse_playlist(itempath)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 755, in _parse_playlist
    playlist_item_id = self._get_item_id(playlist_path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 833, in _get_item_id
    return create_safe_string(file_path.replace(self.config.path, ""))
AttributeError: 'NoneType' object has no attribute 'replace'
2022-07-23 12:17:28 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140159262078096] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 114, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 567, in websocket_playlist_tracks
    for item in await mass.music.playlists.tracks(msg[ITEM_ID], msg[PROVIDER])
  File "/usr/local/lib/python3.10/site-packages/music_assistant/controllers/music/playlists.py", line 45, in tracks
    playlist = await self.get(item_id, provider, provider_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/media_controller.py", line 162, in get
    details = await self.get_provider_item(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/media_controller.py", line 395, in get_provider_item
    item = await provider.get_item(self.media_type, item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 248, in get_item
    return await self.get_playlist(prov_item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 321, in get_playlist
    return await self._parse_playlist(itempath)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 755, in _parse_playlist
    playlist_item_id = self._get_item_id(playlist_path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 833, in _get_item_id
    return create_safe_string(file_path.replace(self.config.path, ""))
AttributeError: 'NoneType' object has no attribute 'replace'
```

**Comment by marcelveldt** [comment]
When did that happen? When browsing ?

Do note that you can enter ALL/MULTIPLE of the channels into that m3u file.

Are you sure that you only have those url's in the m3u file and not something else ?
It's crashing badly as it tries to parse the textline into a filename instead of url

**Comment by jurgenhaas** [comment]
Yes, that file only contains one line `http://192.168.3.54:5004/auto/v301` followed by a line feed. Let me try to put multiple lines into it.

**Comment by jurgenhaas** [comment]
Still the same. The file now contains this:

```
http://192.168.3.54:5004/auto/v301
http://192.168.3.54:5004/auto/v302
http://192.168.3.54:5004/auto/v307
```

When I browse the file system, I see this:

![2022-07-23_12-26](https://user-images.githubusercontent.com/1475675/180601336-5cd521b0-fe35-41d5-8d59-a4df7bd85c8c.png)

When I click on `channels`, the exception appears in the log file and the result looks like this:

![2022-07-23_12-27](https://user-images.githubusercontent.com/1475675/180601358-7b8be050-a723-43ce-bd6c-146f14c6876b.png)

**Comment by marcelveldt** [comment]
OK, I'll quickly test this myself as this should work. There are multiple users that have m3u playlists with just all kind of radio streams in there.

to be continued...

**Comment by marcelveldt** [comment]
<img width="819" alt="Schermafbeelding 2022-07-23 om 15 01 29" src="https://user-images.githubusercontent.com/6389780/180606065-3d5b54d0-5881-4735-ba62-d9338da4c27e.png">

working fine on my end, created an m3u with some http radio streams in it and they're listed fine and streaming also works. Are you on the latest version 2022.7 ?
[test_radiostreams.m3u.txt](https://github.com/music-assistant/hass-music-assistant/files/9173809/test_radiostreams.m3u.txt)

See attached the test playlist file I used. Rename .txt to .m3u again because Gh didn't accept m3u

**Comment by marcelveldt** [comment]
I reproduced the error now. It only happens if you browse to the playlist from the (classic) browse menu.
If you open the playlist from the playlists collection it works fine and once clicked once from the playlists collection, it even works from the browse menu :-)

It's like magic. I'll explore

**Comment by marcelveldt** [comment]
I've just released patch version 2022.7.1 which should fix this issue.
If you still experience any issues after updating to 2022.7.1 please let us know, thanks!

**Comment by jurgenhaas** [comment]
MA is working much better now, it scans all my media which it didn't do before but I haven't analyzed what was wrong before. Some other change must have fixed that, thanks.

But the radio streams still cause issues. I have a channels.m3u file with your 3 URLs in it, nothing else. That throws this exception when browsing into that directory:

```
homeassistant-app-1  | 2022-07-26 08:14:50 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139944736905088] Error handling message: Unknown error (unknown_error)
homeassistant-app-1  | Traceback (most recent call last):
homeassistant-app-1  |   File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
homeassistant-app-1  |     await func(hass, connection, msg)
homeassistant-app-1  |   File "/config/custom_components/mass/websockets.py", line 114, in async_get_mass_func
homeassistant-app-1  |     await orig_func(hass, connection, msg, mass)
homeassistant-app-1  |   File "/config/custom_components/mass/websockets.py", line 929, in websocket_browse
homeassistant-app-1  |     await mass.music.browse(msg.get("path")),
homeassistant-app-1  |   File "/usr/local/lib/python3.10/site-packages/music_assistant/controllers/music/__init__.py", line 224, in browse
homeassistant-app-1  |     return await prov.browse(path)
homeassistant-app-1  |   File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/filesystem.py", line 196, in browse
homeassistant-app-1  |     item_id, provider_id=self.id
homeassistant-app-1  | UnboundLocalError: local variable 'item_id' referenced before assignment
```

It's different from before, so somethings must have changed at least.

**Comment by jurgenhaas** [comment]
Ping @marcelveldt just in case you had not seen my previous comment due to the closed state of this issue.

**Comment by marcelveldt** [comment]
You're right, I do miss comments made on closed issue pretty fast. I'll take another look

**Comment by marcelveldt** [comment]
Found the typo. Fixed in 2022.7.2 !
